crates/sparrow-backend/src/logical_to_physical.rs

```diff
@@ -118,11 +136,21 @@ impl LogicalToPhysical {
                 // of binary merges by choosing the order to perform the merges based on the number
                 // of rows.
                 let inputs: Vec<_> = must_merge.into_iter().collect();
+                let result_fields: arrow_schema::Fields = inputs
+                    .iter()
+                    .map(|input_step| {
+                        self.step_type(*input_step).map(|data_type| {
+                            Field::new(format!("step_{}", input_step), data_type, true)
```

Comment:
I worry about when we have to manually set the nullability. Any reason we think it could ever be `false`?

Reply:
Hm no suppose not. The main problems I had with this was when I had to verify the type and it was getting mad that a nullable vs not field was different. In this case I think it's okay.